src/main/java/util/Engine.java

```diff
@@ -6,73 +6,123 @@
 
 import static org.lwjgl.glfw.GLFW.glfwInit;
 
+/**
+ * The Engine class initializes GLFW and the game loop. <br\>
+ * Singleton.
+ */
 public final class Engine {
-
     /**
-     * The Engine class initializes GLFW and the game loop.
+     * The static {@code Engine} instance.
      */
-
     private static final Engine instance = new Engine();
-
+    
+    /**
+     * The exact time the engine started in milliseconds.
+     */
     private final long startMillis;
+    
+    /**
+     * No clue
+     */
     private Window window;
+    
+    /**
+     * Whether the {@code Engine} is running.
+     */
     private boolean running;
+    
+    /**
+     * The time between last frame and current
+     */
     private float deltaTime;
-
+    
     //private to prevent creating new instances
+    /**
```

Comment:
this is generally the purpose of `private`, there is no need to comment this.

Reply:
the comment was already there, i just made it to a javadoc. will remove it 👍